### ISSUE TITLE
Lazy loading design direction

### ISSUE THREAD

**Issue by brianfeister** [angular/router#162]
Hey @btford!  I tweeted you about this and thought you might like to have it documented here.  

In [this video](https://www.youtube.com/watch?v=vecg70fPDFw&list=PLOETEcp3DkCoNnlhE-7fovYvqwVPrRiY7&t=849) you mention `newRouter` lazy loading. I have [this seed project](http://github.com/brianfeister/angular-systemjs-seed) using systemjs. I'm hoping that I'll get to play along with the very early alpha development of Angular 2 + newRouter because I have a release date some time around Jan 2016. 

My main question is about the scope of your plans for lazy-loading components with newRouter.  Right now, this project uses a `routeBundler` helper that wires all of the lazy-loaded dependency injection via systemjs.  I know this is super early, but will I be able to do this with newRouter?  If so, I would probably spend less time on the systemjs workflow automation with the hope of discarding it in favor of newRouter's dependency injection.

**Comment by gdi2290** [comment]
that's a nice set up

**Comment by jeffwhelpley** [comment]
@brianfeister thanks for putting together the seed project. Does it assume you are not packaging your resources (i.e. they remain as individual files on the server)? 

If so, I would also want to be able to define packages of modules that can be lazy loaded. For example, somehow you define that x, y and z are packaged as part of foo and then do `import {x, y, z} from 'foo'`.

**Comment by brianfeister** [comment]
The project includes an NPM dependency https://github.com/Swimlane/systemjs-route-bundler, which allows you to bundle routes from [routes.jons](https://github.com/brianfeister/angular-systemjs-seed/blob/master/client/src/app/routes.json) and systemjs does the rest.  The project has a bit of weirdness which I'm trying to iron out, feel free to PR any fixes you end up making.  It works, but it's not quite "clean" yet.

**Comment by jeffwhelpley** [comment]
I am still looking through all the router issues and found this one talks about the potential ability to do what I am thinking with 1.5: https://github.com/angular/router/issues/22#issuecomment-73576038

**Comment by brianfeister** [comment]
This is relevant as well: https://twitter.com/briantford/status/574984205404401664

**Comment by btford** [comment]
> will I be able to do this with newRouter?

Yes, I have similar goals. I plan to keep the routing system static enough that you can figure this information out statically so your code loader knows how to put your app together.

More on this soon!

**Comment by csbenjamin** [comment]
+1

**Comment by MartinDesigner** [comment]
![imghome](https://cloud.githubusercontent.com/assets/13809960/9289329/0bef602c-436c-11e5-80fc-3ec07f82dd8d.jpg)
So, as a supplement, however as a main point try this amazing free app. Createer.com is an online software for easily designing online ads, responsive landing pages and print materials such as flyers, posters, covers, invitations and more. Createer lets you easily share your designs on social media, send them via email, download as HTML, image or PDF. You can even design UI mockups for mobile as for standard display (html, js, css always generating without any coding required from you). Createer also includes 3,000 predesigned templates (browse here: https://beta2.createer.com/Templates?ghi) and over 250,000 images (illustrations, photos and icons) to work with in its user-friendly drag’n’drop enviroment. And finally this amazing stuff is here: https://beta2.createer.com/?ghi
